test/system/030-run.bats

```diff
@@ -1497,6 +1497,50 @@ EOF
     done
 }
 
+# bats test_tags=ci:parallel
+@test "podman run --restart preserves hooks-dir" {
+    # regression test for #17935 to ensure hooks are run on successful restarts
+    ctr=c_$(safename)
+    hooksdir=$PODMAN_TMPDIR/hooks_$(safename)
+
+    skip_if_remote "--hooks-dir is not usable with remote"
+
+    mkdir -p "$hooksdir"
+    cat > "$hooksdir/settings.json" <<EOF
+{
+    "version": "1.0.0",
+    "when": { "always": true },
+    "hook": {
+        "path": "$hooksdir/hook.sh"
+    },
+    "stages": ["prestart", "poststop"]
+}
+EOF
+    cat >"$hooksdir/hook.sh" <<EOF
+#!/bin/sh
+# consume stdin
+cat > /dev/null
+echo ran >> "$hooksdir/log"
+EOF
+    chmod +x "$hooksdir/hook.sh"
+
+    run_podman run -d --restart=on-failure:2 \
+        --name="$ctr" --hooks-dir="$hooksdir" $IMAGE false
+
+    wait_for_restart_count "$ctr" 2 "restart preserves hooks-dir"
+
+    # also make sure 3rd restart has finished
+    # wait also waits until 'cleanup' is done, so poststop hook is
+    # done running after this command
+    run_podman wait "$ctr"
+    run_podman rm "$ctr"
+
+    # check prestart and poststop ran 3 times each
+    assert "$(wc -l < "$hooksdir/log")" = 6
+
+    rm -r "$hooksdir"
```

Comment:
$PODMAN_TMPDIR is unique for each test and automatically removed so technically there should be no need for this. But doing this is of course not a problem so we can keep it unless you need to repush for other reasons.